pkg/component/etcd/service/service_peer.go

```diff
@@ -17,19 +17,20 @@ package service
 import (
 	"context"
 
+	"github.com/gardener/etcd-druid/pkg/utils"
 	"github.com/gardener/gardener/pkg/controllerutils"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
 func (c *component) syncPeerService(ctx context.Context, svc *corev1.Service) error {
 	_, err := controllerutils.GetAndCreateOrStrategicMergePatch(ctx, c.client, svc, func() error {
-		svc.Labels = getLabels(c.values)
+		svc.Labels = utils.MergeStringMaps(getLabels(c.values), getSelectors(c.values))
```

Comment:
Since `"instance": val.EtcdName` is now present as part of `getSelectors()`, it can be removed from `getLabels()` [here](https://github.com/gardener/etcd-druid/blob/master/pkg/component/etcd/service/service.go#L101-L111)